diesel/src/mysql/connection/raw.rs

```diff
@@ -46,10 +46,10 @@ impl RawConnection {
             // Make sure you don't use the fake one!
             ffi::mysql_real_connect(
                 self.0,
-                host.map(|x| x.as_ptr()).unwrap_or(ptr::null_mut()),
+                host.map(CStr::as_ptr).unwrap_or_else(|| ptr::null_mut()),
```

Comment:
What's the lint that led to the `unwrap_or_else` change? `ptr::null_mut` is a `const fn` so it probably shouldn't be complaining here.